src/services/refactors/convertParamsToDestructuredObject.ts

```diff
@@ -51,18 +51,14 @@ namespace ts.refactor.convertParamsToDestructuredObject {
         changes: textChanges.ChangeTracker,
         functionDeclaration: ValidFunctionDeclaration,
         groupedReferences: GroupedReferences): void {
-        const newParamDeclaration = map(createNewParameters(functionDeclaration, program, host), param => getSynthesizedDeepClone(param));
-        changes.replaceNodeRangeWithNodes(
-            sourceFile,
-            first(functionDeclaration.parameters),
-            last(functionDeclaration.parameters),
-            newParamDeclaration,
-            {   joiner: ", ",
-                // indentation is set to 0 because otherwise the object parameter will be indented if there is a `this` parameter
-                indentation: 0,
-                leadingTriviaOption: textChanges.LeadingTriviaOption.IncludeAll,
-                trailingTriviaOption: textChanges.TrailingTriviaOption.Include
-            });
+        const signature = groupedReferences.signature;
+        const newFunctionDeclarationParams = map(createNewParameters(functionDeclaration, program, host), param => getSynthesizedDeepClone(param));
+
+        if (signature) {
+            const newSignatureParams = map(createNewParameters(signature, program, host), param => getSynthesizedDeepClone(param));
+            replaceParameters(signature, newSignatureParams);
+        }
+        replaceParameters(functionDeclaration, newFunctionDeclarationParams);

```

Comment:
Should these branches be mutually exclusive? This looks suspicious to me.

Reply:
If the signature is defined, that means we were able to connect the method declaration back to it. That is, `functionDeclaration` implements `signature` and both should be changed. The absence of `signature` simply means `functionDeclaration` does not implement a method declared elsewhere (that we know of).